test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+// returns the cpushares or error if no such annotation exists.
+func (c *crioContainerData) getAnnotationCPUShare() (int, error) {
+	anno := fmt.Sprintf("%s/%s", WorkloadAnnotationPrefix, c.ContainerName)
+	cpuShare, ok := c.ContainerAnnotations[anno]
+	if !ok {
+		return 0, fmt.Errorf("workload annotation of [%s] was expected but not found", anno)
+	}
+
+	// workload cpushare annotations have a json string that will contain `{ "cpushares": <share int> }`
+	// we parse and return that value here.
+	temp := map[string]int{}
+	err := json.Unmarshal([]byte(cpuShare), &temp)
+	if err != nil {
+		return 0, fmt.Errorf("err parsing cpushare json annotation: %w", err)
+	}
+
+	value, ok := temp["cpushares"]
+	if !ok {
+		return 0, fmt.Errorf("err cpushare annotation was incorrect expected format {\"cpushares\": <int> }")
+	}
+
+	return value, nil
+}
+
+var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning node validation", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                      = exutil.NewCLIWithoutNamespace("cpu-partitioning").AsAdmin()
+		ctx                     = context.Background()
+		isClusterCPUPartitioned = false
+	)
+
+	g.BeforeEach(func() {
+		isClusterCPUPartitioned = getCpuPartitionedStatus(oc) == ocpv1.CPUPartitioningAllNodes
+	})
+
+	g.It("crio containers should have correct cpuset and cpushare set", func() {
+		nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred(), "error listing cluster nodes")
+
+		nodeRoleCPUSet, err := getExpectedCPUSetConstraints(ctx, oc, isClusterCPUPartitioned)
+		o.Expect(err).ToNot(o.HaveOccurred(), "error getting node CPUSets")
+
+		g.By(fmt.Sprintf("using nodeRoleCPUSet %s", nodeRoleCPUSet))
+
+		for _, node := range nodes.Items {
+			// Collect the container data from the node
+			crioData, err := collectContainerInfo(oc.AdminConfig(), oc.AsAdmin().KubeFramework().ClientSet, node)
+			o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error getting crio container data from node %s", node.Name))
+
+			// When the cluster is CPU Partitioned there should always be pods with workload annotations. However,
+			// we do not enforce this check on non partitioned clusters since we don't expect nodes beyond masters to have pods
+			// that contain workload annotations. In the event that they do, we allow the logic to continue to validate
+			// CPUSets.
+			if isClusterCPUPartitioned {
+				o.Expect(crioData).ToNot(o.BeEmpty(), "error getting crio container data, container data should not be empty")
+			}
+
+			// CPU Core counts are numbers 0-n, we minus one since Status.Capacity.Cpu count is numbered conventionally.
+			// We generate a wide open CPUSet to use for comparison.
+			totalCoreCount := node.Status.Capacity.Cpu().Value() - 1
+			parsedFullNodeCPUSet, err := cpuset.Parse(fmt.Sprintf("0-%d", totalCoreCount))
```

Comment:
I was wondering if a plain `for` loop and then using `cpuset.New` is better, but in this context it probably your approach is fine

Reply:
That's a good point, I wasn't entirely 100% sure on how many different ways the Capacity value was going to come through, so I didn't want to cast it to an `int`. But I don't think there's harm in that, if anything the test will break so that's desired. I'll go ahead and make the update